nodes/manual-seal/src/service.rs

```diff
@@ -16,133 +20,163 @@ native_executor_instance!(
 	runtime::native_version,
 );
 
-/// Starts a `ServiceBuilder` for a full service.
-///
-/// Use this macro if you don't actually need the full service, but just the builder in order to
-/// be able to perform chain operations.
-macro_rules! new_full_start {
-	($config:expr) => {{
-		let builder = sc_service::ServiceBuilder::new_full::<
-			runtime::opaque::Block,
-			runtime::RuntimeApi,
-			crate::service::Executor,
-		>($config)?
-		.with_select_chain(|_config, backend| Ok(sc_consensus::LongestChain::new(backend.clone())))?
-		.with_transaction_pool(|builder| {
-			let pool_api = sc_transaction_pool::FullChainApi::new(builder.client().clone());
-			Ok(sc_transaction_pool::BasicPool::new(
-				builder.config().transaction_pool.clone(),
-				std::sync::Arc::new(pool_api),
-				builder.prometheus_registry(),
-			))
-		})?
-		.with_import_queue(
-			|_config, client, _select_chain, _transaction_pool, spawn_task_handle, registry| {
-				Ok(sc_consensus_manual_seal::import_queue(
-					Box::new(client),
-					spawn_task_handle,
-					registry,
-				))
-			},
-		)?;
-
-		builder
-		}};
-}
-
-type RpcExtension = jsonrpc_core::IoHandler<sc_rpc::Metadata>;
-
-/// Builds a new service for a full client.
-pub fn new_full(config: Configuration) -> Result<impl AbstractService, ServiceError> {
-	let is_authority = config.role.is_authority();
-
+type FullClient = sc_service::TFullClient<Block, RuntimeApi, Executor>;
+type FullBackend = sc_service::TFullBackend<Block>;
+type FullSelectChain = sc_consensus::LongestChain<FullBackend, Block>;
+
+/// Returns most parts of a service. Not enough to run a full chain,
+/// But enough to perform chain operations like purge-chain
+pub fn new_full_params(config: Configuration) -> Result<(
+	sc_service::ServiceParams<
+		Block, FullClient,
+		BasicQueue<Block, TransactionFor<FullClient, Block>>,
+		sc_transaction_pool::FullPool<Block, FullClient>,
+		(), FullBackend,
```

Comment:
This needs to be the output of `crate::rpc::create_full`:
```suggestion
		jsonrpc_core::io::IoHandler, FullBackend,
```